src/librawspeed/decompressors/AbstractLJpegDecompressor.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+    RawSpeed - RAW file decoder.
+
+    Copyright (C) 2009-2014 Klaus Post
+    Copyright (C) 2017 Axel Waggershauser
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#include "decompressors/AbstractLJpegDecompressor.h"
+#include "common/Common.h"                // for uint32, getHostEndianness
+#include "common/Point.h"                 // for iPoint2D
+#include "decoders/RawDecoderException.h" // for ThrowRDE
+#include "decompressors/HuffmanTable.h"   // for HuffmanTable
+#include "io/ByteStream.h"                // for ByteStream
+#include "io/IOException.h"               // for IOException
+#include <array>                          // for array
+#include <memory>                         // for unique_ptr, allocator
+#include <vector>                         // for vector
+
+namespace RawSpeed {
+
+void AbstractLJpegDecompressor::decode() {
+  if (getNextMarker(false) != M_SOI)
+    ThrowRDE("LJpegDecompressor: Image did not start with SOI. Probably not an LJPEG");
```

Comment:
Would LJpegDecompressor in the throw expression message have to be adjusted to AbstractLJpegDecompressor?